### PR TITLE
New implementation.

### PR DESCRIPTION
**Pull request by dabrahams**
Tests need updating:

- ensure that ADOBE_CONTRACT_VIOLATION=minimal doesn't cause verbose output
- ensure that misconfiguration produces a good message

### CODE REVIEW

File: include/adobe/contract_checks.hpp [hunk 1]
```diff
@@ -1,169 +1,117 @@
...
-#ifdef ADOBE_CONTRACT_CHECKS_CONFIGURATION
-#include ADOBE_CONTRACT_CHECKS_CONFIGURATION
-#endif
-
-#ifdef ADOBE_CONTRACT_VIOLATED_THROWS
-#define INTERNAL_ADOBE_CONTRACT_VIOLATED_NOEXCEPT
+// INTERNAL_ADOBE_CONTRACT_VIOLATION_BEHAVIOR is not a function-style
+// macro because it lets us give a better diagnostic on
+// misconfiguration.
+#ifndef ADOBE_CONTRACT_VIOLATION// Default is verbose
```
Comment (sean-parent) [inline include/adobe/contract_checks.hpp:11]:
```suggestion
#ifndef ADOBE_CONTRACT_VIOLATION // Default is verbose
```

**Review by sean-parent — APPROVED** [review]
LGTM - I'm assuming subsequent PRs to update the readme.
Do we want a macro to query the configuration?

```cpp
#if !ADOBE_CONTRACT_CONFIGURATION(unsafe)
int _uncaught_count = std::uncaught_exceptions();
#endif
...
```

File: include/adobe/contract_checks.hpp (hunk 1 above)
Comment (dabrahams) [reply include/adobe/contract_checks.hpp:11]: You can do that, but clang-format will put it back to the way it is.  If you want to change that you need to update the formatting file.

**Comment by dabrahams** [comment]
> Do we want a macro to query the configuration?

You tell **me**!  Do you have a use-case?

File: include/adobe/contract_checks.hpp (hunk 1 above)
Comment (sean-parent) [reply include/adobe/contract_checks.hpp:11]: Okay - I'll look at updating the format file (the current one is horrible).

**Comment by sean-parent** [comment]
> > Do we want a macro to query the configuration?
> 
> You tell **me**! Do you have a use-case?

The primary use case is building additional facilities either on top of the current ones or alongside. Yes. We want to be able to query.

**Comment by dabrahams** [comment]
> Yes. We want to be able to query.

Can you open an issue?  Definitely a separable feature and I'd like to land a v0.9

**Comment by dabrahams** [comment]
@fosterbrereton Did you really mean to delete this branch and close my PR?

**Pull request merged**